File: MiniDOMWebCtrls/DMD/Anagrafica/RelazioniParentaliHandler.vb
Imports minidom
Imports minidom.Sistema
Imports minidom.Databases
Imports minidom.WebSite
Imports minidom.Anagrafica
Imports minidom.Forms.Utils


Imports minidom.Web
Imports minidom.XML

Namespace Forms

    Public Class RelazioniParentaliHandler
        Inherits CBaseModuleHandler

        Public Sub New()
            MyBase.New(ModuleSupportFlags.SAnnotations Or ModuleSupportFlags.SCreate Or ModuleSupportFlags.SDelete Or ModuleSupportFlags.SDuplicate Or ModuleSupportFlags.SEdit)
            
        End Sub

        Public Overrides Function CreateCursor() As DBObjectCursorBase
            Return New CRelazioneParentaleCursor
        End Function


        Public Function GetRelazioni(ByVal renderer As Object) As String
            Dim pID As Integer = RPC.n2int(Me.GetParameter(renderer, "pid", ""))
            Dim items As CCollection(Of CRelazioneParentale) = Anagrafica.RelazioniParentali.GetRelazioni(pID)
            Return XML.Utils.Serializer.Serialize(items)
        End Function

    End Class



End Namespace

File: MiniDOMWebCtrls/DMD/Anagrafica/RelazioniParentaliHandler.cs

namespace minidom.Forms
{
    public class RelazioniParentaliHandler : CBaseModuleHandler
    {
        public RelazioniParentaliHandler() : base(ModuleSupportFlags.SAnnotations | ModuleSupportFlags.SCreate | ModuleSupportFlags.SDelete | ModuleSupportFlags.SDuplicate | ModuleSupportFlags.SEdit)
        {
        }

        public override Databases.DBObjectCursorBase CreateCursor()
        {
            return new Anagrafica.CRelazioneParentaleCursor();
        }

        public string GetRelazioni(object renderer)
        {
            int pID = (int)this.n2int(renderer, "pid");
            var items = Anagrafica.RelazioniParentali.GetRelazioni(pID);
            return DMD.XML.Utils.Serializer.Serialize(items);
        }
    }
}